src/status_im/chat/events.cljs

```diff
@@ -75,14 +83,41 @@
   (fn [db [{:keys [chat-id message-id]}]]
     (update-in db [:chats chat-id :messages message-id] assoc :appearing? false)))
 
+(handlers/register-handler-fx
+  :set-message-envelope-hash
+  [re-frame/trim-v]
+  (fn [{:keys [db]} [chat-id message-id envelope-hash]]
+    {:db (assoc-in db [:message-envelopes envelope-hash] {:chat-id    chat-id
```

Comment:
Can we hold this in already existing`:transport/chats` db subpath like this:
`{:transport/chats {chat-id {:envelope->message-id envelope-hash message-id}}}`
Otherwise we will again end up with a situation we just fixed not so long ago, where every thing related to messages (it was `:message-data`, `:message-statuses` before) will have it's own place in db and keeping it sane and in sync will be difficult.

Reply:
@janherich not sure we could construct such path as later on signal would be sent from status-go which provides envelope hash only and no chat-id (i'm talking about `:signals/envelope-status` event)

Reply:
You are right, we need envelope hash as the primary key (index) in this case unfortunately.
I would at least rename the path so it will be `:transport/message-envelopes` to signify it's very much transport related.

Reply:
Another observation:

at the moment we allow users to leave the chat which will delete the `transport/chat->chat-id`, at that point `:signals/envelope-status` would not behave as expected if we received a signal for a message from a chat we just left. 
Probably not worth worrying about as an edge case, but worth pointing out.

Reply:
following on @janherich  suggestion, if we have also message-id we can make the code more resilient & less statefull by not assuming that the message is in the re-frame db. So instead of having:

1) Receive signal
2) Get message from db
3) Update realm

You could have

1) Receive signal
2) Update realm
3) Update re-frame db if message is there (update single field)

Which is stronger as we make less assumptions (only that the message is in the real-db, but you can loosen that one if we want).

But yeah, not important for now, as we don't offload messages from the re-frame database.